invokeai/app/services/invocation_cache/invocation_cache_memory.py

```diff
@@ -0,0 +1,46 @@
+from queue import Queue
+from typing import Optional, Union
+
+from invokeai.app.invocations.baseinvocation import BaseInvocation, BaseInvocationOutput
+from invokeai.app.services.invocation_cache.invocation_cache_base import InvocationCacheBase
+
+
+class MemoryInvocationCache(InvocationCacheBase):
```

Comment:
nit: We could convert this from a FIFO cache to a LRU cache pretty easily with an OrderedDict, which would probably be a little better. (sample implementation: https://www.geeksforgeeks.org/lru-cache-in-python-using-ordereddict/)

I don't think it make sense to hold things up for this, so just noting for future.

Reply:
Sounds good!